token/link/link.go

```diff
@@ -205,6 +209,7 @@ func (service *LinkService) Callback(ctx context.Context, req *goa.RequestData,
 			// It was re-linking. Overwrite the existing link.
 			externalToken := tokens[0]
 			externalToken.Token = providerToken.AccessToken
+			externalToken.Username = userProfile.Username
```

Comment:
so reconnecting is how the username will be updated.

Reply:
Yes.